lib/weatherApp.js

```diff
@@ -0,0 +1,46 @@
+(function (factory) {
+    if (typeof define === 'function' && define['amd']) {
+        define(['knockout'], factory);
+    } else if (typeof require === 'function' && typeof exports === 'object' && typeof module === 'object') {
+        module.exports = factory(require('knockout'));
+    } else {
+        window.WeatherApp = factory(ko);
+    }
+} (function (ko) {
+    return function WeatherApp() {
+        'use strict';
+        let city = ko.observable("");
+        let api = "http://api.openweathermap.org/data/2.5/weather?q=";
+        let unit = "&units=metric&";
+        let apiId = "appid=bd959553bce6759da749a8bcad48f038";
+        let maxTemp = ko.observable("");
+        let minTemp = ko.observable("");
+        let isTempVisible = ko.observable(false);
+        let josnCityName = ko.observable("");
+
+        function showWeather() {
+            let url = api + city() + unit + apiId;
+            fetch(url)
+                .then((response => {
+                    response.json()
+                        .then(josn => {
```

Comment:
I don't see any sort of error handling here. For example, if I type "@@@@@" or "zzzzzzzz" and press submit, nothing happens. It would be nice if it could display an error message saying "Could not find the city @@@@@". Also if I don't have an internet connection and I press Submit then nothing happens. It would be good if it said something like "Couldn't get weather information. Please check your internet connection."